### DISCUSSION TITLE
Project repository & ModelSubmitForm

### DISCUSSION THREAD

**Discussion by lehors** [lfai/model_openness_tool#32] [General]
For non-admin users, the GitHub address of a model is set by selecting among the user's repos. I don't think that works.

First it assumes that the user owns the model repo which is clearly not going to be the true in many cases. Then that list can be arbitrary long. In my personal case, I currently have over 75 repos (mostly clones of projects I worked on at one point or another) and the list seems to be truncated at the first 20 or so.

@gregcube Can you please share some light as to what the rationale for this was?

I think we'd be better served with a simple text field but I can see quite a bit of effort was put into this functionality and I hate to just throw it out.

**Comment by gregcube** [comment]
I believe the original idea was to streamline model input, assuming that the github user is indeed the model repository owner.  Selecting a repository automatically populates the "Name" and "Description" fields. Additionally, when a component license is entered, the "License path" and "Component path" fields are linked to a hierarchical list of the repository's contents. As you type, it provides full path suggestions to the license file or component.

> I think we'd be better served with a simple text field but I can see quite a bit of effort was put into this functionality and I hate to just throw it out.

I'm thinking this as well, given that we can no longer assume the github user is the model owner.  I'm fine with a simple text field replacing this bit.  With a text field (under "Model details" perhaps?) we can likely still capture the repository's filesystem tree to allow easy completion of the license file or component path.  If the repository is public, the API endpoint will still return the git tree object even if the user is not the owner.

**Reply by lehors** [reply]
Thank you for the background info. This is very helpful.

> I believe the original idea was to streamline model input, assuming that the github user is indeed the model repository owner. Selecting a repository automatically populates the "Name" and "Description" fields. Additionally, when a component license is entered, the "License path" and "Component path" fields are linked to a hierarchical list of the repository's contents. As you type, it provides full path suggestions to the license file or component.

I had not even realized that all of this was in there! That's actually pretty neat.

> 
> > I think we'd be better served with a simple text field but I can see quite a bit of effort was put into this functionality and I hate to just throw it out.
> 
> I'm thinking this as well, given that we can no longer assume the github user is the model owner. I'm fine with a simple text field replacing this bit. With a text field (under "Model details" perhaps?) we can likely still capture the repository's filesystem tree to allow easy completion of the license file or component path. If the repository is public, the API endpoint will still return the git tree object even if the user is not the owner.

Indeed, it sounds like we'd end up with the best of both worlds this way. I like it.

I think we should have the text field first in the Model details section so that the automatic population of the other fields can take place before the user gets to those fields. I don't know how much time this takes though. Is the UI on hold while this takes place? We wouldn't want to have the user start typing the name to see it being replaced by what is pulled from the repo.
